app/models/feeds/apple_subscription.rb

```diff
@@ -14,14 +13,15 @@ class Feeds::AppleSubscription < Feed
   validate :unchanged_defaults
   validate :only_apple_feed
   validate :must_be_private
+  validate :must_have_token
```

Comment:
This reads really nicely and is a good addition!

A couple of questions / just thinking out loud:

In what context do you expect this validation error/flag to be triggered? Something like a future code refactoring?

Assuming we're in the "reject" state for this validation (somehow the apple subscription no longer has a token), how must we test to ensure we surface the validation message?

Reply:
q1: yes, refactoring is probably the context i would expect to see this. like if suddenly we allowed users to add tokens to apple feeds, and changed around the UI to be more like current private feeds (where the tokens array can be empty), hopefully this validation would catch that. i don't expect this validation to show up in the current UI since users aren't responsible for adding tokens to apple feeds.

q2: i'm not quite sure i understand the question. is this related to q1, like where does this validation show up? if so, then same answer, doesn't need to be surfaced in the current UI; for developer use only, at the moment.